src/psyclone/tests/psyir/backend/fortran_test.py

```diff
@@ -637,3 +637,26 @@ def test_fw_codeblock():
         "    a=1\n"
         "PRINT *, 'I am a code block'\n"
         "    PRINT *, 'with more than one line'\n" in result)
+
+def test_fw_loop():
+    '''Check the FortranWriter class loop method
+    correctly prints out the Fortran representation.
+
+    '''
+    # Generate fparser2 parse tree from Fortran code.
+    code = (
+        "module test\n"
+        "contains\n"
+        "subroutine tmp()\n"
+        "  integer :: i, sum\n"
```

Comment:
Strictly speaking we should ensure `sum` is initialised.

Reply:
done